### PR TITLE
V0 39 add integration

### PR DESCRIPTION
**Pull request by StanleyCuenat**
sorry for the big pull request....

### CODE REVIEW

File: src/Dao/Models/User/User.saga.ts [hunk 1]
```diff
@@ -89,6 +89,12 @@ function* storeUserRequest(action: UserType.IStoreUserAction) {
         yield put(UserAction.getUserFail())
     }
 }
+
+export function* patchUserInformation(form: any) {
+    yield delay(1000)
```
Comment (Aksamyt) [inline src/Dao/Models/User/User.saga.ts]: 👀

File: src/View/Components/Button/ToolButton/ToolButton.tsx [hunk 2]
```diff
@@ -17,18 +17,48 @@
...
+                return 'h-24 w-24'
+            case 'xl':
+                return 'h-40 w-40'
+            default:
+                return 'h-16 w-16'
+        }
+    }
+
+    isActive = () => {
+        return this.props.active === true
```
Comment (Aksamyt) [inline src/View/Components/Button/ToolButton/ToolButton.tsx]: 👀

File: src/View/Components/OnBoarding/Info/Info.tsx [hunk 3]
```diff
@@ -18,54 +18,59 @@
...
+        document.body.removeAttribute(script.id)
     }
-    errorLocation = () => {
-        const { t } = this.props
-        toast.error(t('user:info.toaster.error.geo_location'))
-        this.setState({ isLocalizing: false })
+
+    searchPlace = async (str: string) => {
+        var geocoder = new maptiler.Geocoder({
+            key: 'GWHPURfVTsxh57TnbRgP',
```
Comment (Aksamyt) [inline src/View/Components/OnBoarding/Info/Info.tsx]: 👀

File: src/View/Components/OnBoarding/Info/Info.type.ts [hunk 4]
```diff
@@ -20,11 +20,12 @@ import { WithTranslation } from 'react-i18next'
 import * as UserAction from 'Dao/Models/User/User.action'
 
 export interface IProps extends WithTranslation {
-    history: any
-    storeUserRequest: typeof UserAction.storeUserRequest
+    // storeUserRequest: typeof UserAction.storeUserRequest
```
Comment (Aksamyt) [inline src/View/Components/OnBoarding/Info/Info.type.ts]: Remove the comment

File: src/View/Pages/Profile/Tools/Tools.tsx [hunk 5]
```diff
@@ -65,13 +65,14 @@ class Tools extends Component<Types.IProps, Types.IState> {
...
-                {toolsList.map((tool: ITool) => (
+                {/* {toolsList.map((tool: ITool) => (
                     <ToolButton
                         label={tool.name}
                         onClick={() => console.log('ACTION')}
                         key={tool.name.toLowerCase()}
+                        disabled={false}
                     />
-                ))}
+                ))} */}
```
Comment (Aksamyt) [inline src/View/Pages/Profile/Tools/Tools.tsx]: Remove the comment

File: src/View/Pages/Profile/Tools/Tools.tsx [hunk 6]
```diff
@@ -101,7 +102,7 @@ class Tools extends Component<Types.IProps, Types.IState> {
                             </Fragment>
                         )}
                     </Fragment>
-                )}
+                )} */}
```
Comment (Aksamyt) [inline src/View/Pages/Profile/Tools/Tools.tsx]: Remove the comment

File: src/Dao/Models/User/User.saga.ts (hunk 1 above)
Comment (Aksamyt) [inline src/Dao/Models/User/User.saga.ts]: 👀

File: src/View/Components/StepPoint/StepPoint.tsx [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import React, { Component } from 'react'
+import MeteorSvg from 'Resources/images/icons/meteor.svg'
+import * as Types from './StepPoint.type'
+import './StepPoint.scss'
+export default class StepPoint extends Component<Types.IProps, Types.IState> {
+    render() {
+        return (
+            <div className="step-point flex text-xs">
+                {this.props.point}
+                <img src={MeteorSvg} alt="" />
```
[the remark this answers is not in the document]
Comment (StanleyCuenat) [reply src/View/Components/StepPoint/StepPoint.tsx]: yes but actually it's not use as Component but as image src (it mean that its actually a string)

File: src/View/Components/OnBoarding/Invite/Invite.tsx [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                return {
+                    emailOne: '',
+                    emailTwo: '',
+                    emailThree: '',
+                }
+            },
+            mapPropsToStatus: () => {
+                return {}
+            },
+            validationSchema: Yup.object({}),
```
[the remark this answers is not in the document]
Comment (StanleyCuenat) [reply src/View/Components/OnBoarding/Invite/Invite.tsx]: email are optional so we don't need any validation schema

File: src/View/Components/Stepper/Step/Step.scss [hunk 9]
```diff
@@ -46,13 +44,8 @@
         flex-grow: 0;
     }
     .toolTip {
-        margin: 0 auto;
-        padding: 4px;
-        border-radius: 4px;
         background-image: linear-gradient(113deg, #232a50, #d22f52 100%);
```
Comment (QuentinGaillardSoluti) [inline src/View/Components/Stepper/Step/Step.scss]: The best would be to use variables setted in variable.css for the colors (There is already one blue as example)

**Pull request merged**